### PR TITLE
cli: link to rule names to capa rules website

### LINKED ISSUE #2319: rule app: generate redirects for non-normalized rule names
We uniquely identify capa rules by their filename, which is *mostly* lowercasing and replacing spaces with dashes. but there are a few other rules (https://github.com/mandiant/capa/blob/c3f24c2f48d1119f01f7cf32f9326619f9c9afa9/scripts/lint.py#L109-L116). external services might not know how to generate these rule IDs, and might just want to use the name of the rule, from `rule.meta.name`. We should make it easy for them to generate working rule URLs that end up showing the rule content.

When emitting the HTML file for a rule, emit a second HTML file with name `[verbatim rule name from rule.meta.name].html` whose only content should be the minimal HTML to meta-refresh to the correct URL. This will cause the browser to land the user at the correct URL.

For example, the following URLs should work:
  - https://mandiant.github.io/capa/rules/compiled to the .NET platform (as typed into the browser bar, which gets encoded as...)
  - https://mandiant.github.io/capa/rules/compiled%20to%20the%20.NET%20platform

### LINKED ISSUE #2337: add links from CLI output to capa rules website
we can use a modern terminal feature to make some text a link to a URL, such as described here: https://www.willmcgugan.com/blog/tech/post/real-working-hyperlinks-in-the-terminal-with-rich/

So, for example, each rule name in the default output could be a link to the rules website, showing the rule content.

I think that for terminals that don't support this feature, the link degrades to text, just like is shown today.

The `rich` library that we use for rendering already has a link feature that we can use here.

### PR DESCRIPTION
**Pull request by williballenthin**
closes #2337 
closes #2319

Add modern terminal markup to hyperlink to resources on the web, like entries on the capa rules website.

To make this possible, use rich instead of tabulate to render tables.

image

Also update the canonical path to rules on the website, to: mandiant.github.io/rules/[rule name]/

Using the rule name from the rule header makes it very easy to generate links to the web pages, rather than using the filename of the rule. 

image

### Checklist

- [ ] No CHANGELOG update needed

- [x] No new tests needed

- [x] No documentation update needed

### CODE REVIEW

**Comment by mr-tz** [comment]
related: #2100

**Comment by williballenthin** [comment]
it was really easy to use rich instead of tabulate. i thought there'd be more work but the switch should be painless.

**Comment by williballenthin** [comment]
need to check how redirection to a file is handled

**Comment by williballenthin** [comment]
I've migrated default rendering to rich. It worked well, and I think we can migrate the other renderers, too. But, I'd propose we all try out this PR for a little bit, and if it works, then update the other renderers.

File: web/public/index.html [hunk 1]
```diff
@@ -200,15 +200,15 @@ <h3 class="mt-3">Rule Updates</h3>
...
             overwrite DLL .text section to remove hooks
           </a>
         </li>
 
         <li>
           added:
-          <a href="./rules/attach-bpf-to-socket-on-linux.html">
-            attach BPF to socket on Linux
+          <a href="./rules/attach BPF to socket on Linux/">
+            attach BPF to socket on Linux
```
Comment (mr-tz) [inline web/public/index.html:211]:
```suggestion
            attach BPF to socket on Linux
```

**Review by mr-tz — APPROVED** [review]
looks great, thank you!

File: web/explorer/src/utils/urlHelpers.js [hunk 2]
```diff
@@ -62,9 +62,8 @@ export function createATTACKHref(attack) {
  */
 export function createCapaRulesUrl(node, tag) {
     if (!node || !node.data || !tag) return null;
```
[the remark this answers is not in the document]
Comment (s-ff) [reply web/explorer/src/utils/urlHelpers.js:64]: and remove `tag` param from
https://github.com/mandiant/capa/blob/4133f68573993ab8f8951359d4f116d5b4c4ccab/web/explorer/src/components/RuleMatchesTable.vue#L238

**Review by s-ff — CHANGES_REQUESTED** [review]
Great, this works in my terminal emulator.
Thanks!

File: capa/render/default.py [hunk 3]
```diff
@@ -71,7 +107,12 @@ def rec(match: rd.Match):
     return matches
 
 
-def render_capabilities(doc: rd.ResultDocument, ostream: StringIO):
+def render_rule_name(name: str) -> str:
+    url = f"https://mandiant.github.io/capa/rules/{urllib.parse.quote(name)}/"
```
Comment (s-ff) [inline capa/render/default.py:111]: Clicking the hrefs in the CLI results in 404s:
Current: `https://mandiant.github.io/capa/rules/forwarded export/`
Expected: `https://mandiant.github.io/capa/rules/forwarded-export.html`

I understand this is wip, so just pointing this out for now

**Review by s-ff — APPROVED** [review]

File: capa/render/default.py (hunk 3 above)
Comment (williballenthin) [reply capa/render/default.py:111]: This PR updates the paths that the website will accept. It won't work with the URLs on the website.

Comment (s-ff) [reply capa/render/default.py:111]: Ah, I see

Comment (williballenthin) [reply capa/render/default.py:111]: links work now that the website is updated.

**Comment by s-ff** [comment]
@williballenthin I think you have forgotten to omit `tag` from the `createCapaRulesUrl(node, tag)`.
I also missed it in the second review.

**Pull request merged**